-1

**Failed Tests**:   Build 
**Size**: This PR adds an extra 88KB to repository
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-684b34/52723/summary.html
**COMMIT**: 0cec90ef8c58178a0b68d5f412b9356a2b2b89a1
**CMSSW**: CMSSW_17_0_X_2026-04-17-1100/el8_amd64_gcc13
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/50749/52723/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Failed Build

I found compilation error when building: 

<pre>>> Compiling edm plugin src/DQMServices/FwkIO/plugins/DQMRootOutputModule.cc
/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02937/el8_amd64_gcc13/external/gcc/13.4.0-6908cfdf803923e783448096ca4f0923/bin/c++ -c -DCMS_MICRO_ARCH='x86-64-v3' -DGNU_GCC -D_GNU_SOURCE -DTBB_USE_GLIBCXX_VERSION=130400 -DTBB_SUPPRESS_DEPRECATED_MESSAGES -DTBB_PREVIEW_RESUMABLE_TASKS=1 -DTBB_PREVIEW_TASK_GROUP_EXTENSIONS=1 -DBOOST_SPIRIT_THREADSAFE -DPHOENIX_THREADSAFE -DBOOST_MATH_DISABLE_STD_FPCLASSIFY -DBOOST_UUID_RANDOM_PROVIDER_FORCE_POSIX -DBOOST_MPL_IGNORE_PARENTHESES_WARNING -D__STDC_LIMIT_MACROS -D__STDC_FORMAT_MACROS -DCMSSW_GIT_HASH='CMSSW_17_0_X_2026-04-17-1100' -DPROJECT_NAME='CMSSW' -DPROJECT_VERSION='CMSSW_17_0_X_2026-04-17-1100' -Isrc -Ipoison -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02937/el8_amd64_gcc13/cms/cmssw-patch/CMSSW_17_0_X_2026-04-17-1100/src -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02937/el8_amd64_gcc13/external/classlib/3.1.3-fb14c9d5bf6be85f4454685cbee1e956/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02937/el8_amd64_gcc13/external/pcre/8.43-7f0edb13c32a66555b25b5433dd2ceed/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02937/el8_amd64_gcc13/external/boost/1.80.0-439e0a2183f4d7ca3604168e474224c5/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02937/el8_amd64_gcc13/external/bz2lib/1.0.8-353e2ef1c2f1d06b28a8938defa197bc/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02937/el8_amd64_gcc13/external/libuuid/2.40-d94705b84648fd294d63bf9de23bc03b/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02937/el8_amd64_gcc13/external/protobuf/3.21.9-b07b8f47dd1983d3cd0c0f051c28c6a1/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02937/el8_amd64_gcc13/lcg/root/6.36.11-1e3105afbb5b2411f379b91a3e48a6a5/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02937/el8_amd64_gcc13/external/tbb/v2022.3.0-daad917f8c7fd382a88c95b720f7e5f7/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02937/el8_amd64_gcc13/external/xz/5.6.4-ea99661ac7ca423b4d74fd650284b41f/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02937/el8_amd64_gcc13/external/zlib/1.3.2-53c061acc706fe8f28fd9477d9981c20/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02937/el8_amd64_gcc13/external/fmt/10.2.1-d88182373b289c4f8ed5f2c5e5024c2b/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02937/el8_amd64_gcc13/external/md5/1.0.0-26057075013e190e56dad37d35219376/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02937/el8_amd64_gcc13/external/tinyxml2/6.2.0-ccad39708675ec9caf4eb88d12976cd5/include -O3 -pthread -pipe -Werror=main -Werror=pointer-arith -Werror=overlength-strings -Wno-vla -Werror=overflow -std=c++20 -ftree-vectorize -Werror=array-bounds -Werror=format-contains-nul -Werror=type-limits -fvisibility-inlines-hidden -fno-math-errno --param vect-max-version-for-alias-checks=50 -Xassembler --compress-debug-sections -Wno-error=array-bounds -Warray-bounds -fuse-ld=bfd -march=x86-64-v3 -felide-constructors -fmessage-length=0 -Wall -Wno-non-template-friend -Wno-long-long -Wreturn-type -Wextra -Wpessimizing-move -Wclass-memaccess -Wno-cast-function-type -Wno-unused-but-set-parameter -Wno-ignored-qualifiers -Wno-unused-parameter -Wunused -Wparentheses -Werror=return-type -Werror=unused-value -Werror=unused-label -Werror=address -Werror=format -Werror=sign-compare -Werror=write-strings -Werror=delete-non-virtual-dtor -Werror=strict-aliasing -Werror=narrowing -Werror=unused-but-set-variable -Werror=reorder -Werror=unused-variable -Werror=conversion-null -Werror=return-local-addr -Wnon-virtual-dtor -Werror=switch -fdiagnostics-show-option -Wno-unused-local-typedefs -Wno-attributes -Wno-psabi -Wno-error=unused-variable -DBOOST_DISABLE_ASSERTS -flto=auto -fipa-icf -flto-odr-type-merging -fno-fat-lto-objects -Wodr -fPIC -MMD -MF tmp/el8_amd64_gcc13/src/DQMServices/FwkIO/plugins/DQMServicesFwkIOPlugins/DQMRootOutputModule.cc.d src/DQMServices/FwkIO/plugins/DQMRootOutputModule.cc -o tmp/el8_amd64_gcc13/src/DQMServices/FwkIO/plugins/DQMServicesFwkIOPlugins/DQMRootOutputModule.cc.o
>> Compiling edm plugin src/DQMServices/FwkIO/plugins/DQMRootSource.cc
/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02937/el8_amd64_gcc13/external/gcc/13.4.0-6908cfdf803923e783448096ca4f0923/bin/c++ -c -DCMS_MICRO_ARCH='x86-64-v3' -DGNU_GCC -D_GNU_SOURCE -DTBB_USE_GLIBCXX_VERSION=130400 -DTBB_SUPPRESS_DEPRECATED_MESSAGES -DTBB_PREVIEW_RESUMABLE_TASKS=1 -DTBB_PREVIEW_TASK_GROUP_EXTENSIONS=1 -DBOOST_SPIRIT_THREADSAFE -DPHOENIX_THREADSAFE -DBOOST_MATH_DISABLE_STD_FPCLASSIFY -DBOOST_UUID_RANDOM_PROVIDER_FORCE_POSIX -DBOOST_MPL_IGNORE_PARENTHESES_WARNING -D__STDC_LIMIT_MACROS -D__STDC_FORMAT_MACROS -DCMSSW_GIT_HASH='CMSSW_17_0_X_2026-04-17-1100' -DPROJECT_NAME='CMSSW' -DPROJECT_VERSION='CMSSW_17_0_X_2026-04-17-1100' -Isrc -Ipoison -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02937/el8_amd64_gcc13/cms/cmssw-patch/CMSSW_17_0_X_2026-04-17-1100/src -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02937/el8_amd64_gcc13/external/classlib/3.1.3-fb14c9d5bf6be85f4454685cbee1e956/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02937/el8_amd64_gcc13/external/pcre/8.43-7f0edb13c32a66555b25b5433dd2ceed/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02937/el8_amd64_gcc13/external/boost/1.80.0-439e0a2183f4d7ca3604168e474224c5/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02937/el8_amd64_gcc13/external/bz2lib/1.0.8-353e2ef1c2f1d06b28a8938defa197bc/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02937/el8_amd64_gcc13/external/libuuid/2.40-d94705b84648fd294d63bf9de23bc03b/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02937/el8_amd64_gcc13/external/protobuf/3.21.9-b07b8f47dd1983d3cd0c0f051c28c6a1/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02937/el8_amd64_gcc13/lcg/root/6.36.11-1e3105afbb5b2411f379b91a3e48a6a5/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02937/el8_amd64_gcc13/external/tbb/v2022.3.0-daad917f8c7fd382a88c95b720f7e5f7/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02937/el8_amd64_gcc13/external/xz/5.6.4-ea99661ac7ca423b4d74fd650284b41f/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02937/el8_amd64_gcc13/external/zlib/1.3.2-53c061acc706fe8f28fd9477d9981c20/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02937/el8_amd64_gcc13/external/fmt/10.2.1-d88182373b289c4f8ed5f2c5e5024c2b/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02937/el8_amd64_gcc13/external/md5/1.0.0-26057075013e190e56dad37d35219376/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02937/el8_amd64_gcc13/external/tinyxml2/6.2.0-ccad39708675ec9caf4eb88d12976cd5/include -O3 -pthread -pipe -Werror=main -Werror=pointer-arith -Werror=overlength-strings -Wno-vla -Werror=overflow -std=c++20 -ftree-vectorize -Werror=array-bounds -Werror=format-contains-nul -Werror=type-limits -fvisibility-inlines-hidden -fno-math-errno --param vect-max-version-for-alias-checks=50 -Xassembler --compress-debug-sections -Wno-error=array-bounds -Warray-bounds -fuse-ld=bfd -march=x86-64-v3 -felide-constructors -fmessage-length=0 -Wall -Wno-non-template-friend -Wno-long-long -Wreturn-type -Wextra -Wpessimizing-move -Wclass-memaccess -Wno-cast-function-type -Wno-unused-but-set-parameter -Wno-ignored-qualifiers -Wno-unused-parameter -Wunused -Wparentheses -Werror=return-type -Werror=unused-value -Werror=unused-label -Werror=address -Werror=format -Werror=sign-compare -Werror=write-strings -Werror=delete-non-virtual-dtor -Werror=strict-aliasing -Werror=narrowing -Werror=unused-but-set-variable -Werror=reorder -Werror=unused-variable -Werror=conversion-null -Werror=return-local-addr -Wnon-virtual-dtor -Werror=switch -fdiagnostics-show-option -Wno-unused-local-typedefs -Wno-attributes -Wno-psabi -Wno-error=unused-variable -DBOOST_DISABLE_ASSERTS -flto=auto -fipa-icf -flto-odr-type-merging -fno-fat-lto-objects -Wodr -fPIC -MMD -MF tmp/el8_amd64_gcc13/src/DQMServices/FwkIO/plugins/DQMServicesFwkIOPlugins/DQMRootSource.cc.d src/DQMServices/FwkIO/plugins/DQMRootSource.cc -o tmp/el8_amd64_gcc13/src/DQMServices/FwkIO/plugins/DQMServicesFwkIOPlugins/DQMRootSource.cc.o
src/DQMServices/FwkIO/plugins/DQMRootSource.cc: In member function 'virtual edm::InputSource::ItemTypeInfo DQMRootSource::getNextItemType()':
src/DQMServices/FwkIO/plugins/DQMRootSource.cc:489:74: error: could not convert '((DQMRootSource*)this)->DQMRootSource::m_nextItemType' from 'edm::InputSource::ItemType' to 'edm::InputSource::ItemTypeInfo'
  489 | edm::InputSource::ItemTypeInfo DQMRootSource::getNextItemType() { return m_nextItemType; }
      |                                                                          ^~~~~~~~~~~~~~
      |                                                                          |
      |                                                                          edm::InputSource::ItemType
gmake: *** [tmp/el8_amd64_gcc13/src/DQMServices/FwkIO/plugins/DQMServicesFwkIOPlugins/DQMRootSource.cc.o] Error 1
</pre>

--- a/DQMServices/FwkIO/plugins/DQMRootSource.cc
+++ b/DQMServices/FwkIO/plugins/DQMRootSource.cc
@@ -486,7 +486,7 @@ DQMRootSource::~DQMRootSource() {
 // member functions
 //
 
-edm::InputSource::ItemTypeInfo DQMRootSource::getNextItemType() { return m_nextItemType; }
+edm::InputSource::ItemTypeInfo DQMRootSource::getNextItemType() { return ItemTypeInfo(m_nextItemType); }
 
 // We will read the metadata of all files and fill m_fileMetadatas vector
 std::shared_ptr<edm::FileBlock> DQMRootSource::readFile_() {

--- a/DQMServices/StreamerIO/plugins/DQMProtobufReader.cc
+++ b/DQMServices/StreamerIO/plugins/DQMProtobufReader.cc
@@ -49,23 +49,23 @@ edm::InputSource::ItemTypeInfo DQMProtobufReader::getNextItemType() {
 
     if (edm::shutdown_flag.load()) {
       fiterator_.logFileAction("Shutdown flag was set, shutting down.");
-      return InputSource::ItemType::IsStop;
+      return InputSource::ItemTypeInfo::isStop();
     }
 
     // check for end of run file and force quit
     if (flagEndOfRunKills_ && (fiterator_.state() != State::OPEN)) {
-      return InputSource::ItemType::IsStop;
+      return InputSource::ItemTypeInfo::isStop();
     }
 
     // check for end of run and quit if everything has been processed.
     // this is the clean exit
     if ((!fiterator_.lumiReady()) && (fiterator_.state() == State::EOR)) {
-      return InputSource::ItemType::IsStop;
+      return InputSource::ItemTypeInfo::isStop();
     }
 
     // skip to the next file if we have no files openned yet
     if (fiterator_.lumiReady()) {
-      return InputSource::ItemType::IsLumi;
+      return InputSource::ItemTypeInfo::isLumi();
     }
 
     fiterator_.delay();
@@ -73,7 +73,7 @@ edm::InputSource::ItemTypeInfo DQMProtobufReader::getNextItemType() {
     // IsSynchronize state
     //
     // comment out in order to block at this level
-    // return InputSource::ItemType::IsSynchronize;
+    // return InputSource::ItemTypeInfo::isSynchronize();
   }
 
   // this is unreachable

